File: backend/internal/models/task.go
package models

import (
	"time"
)

type Task struct {
	BaseModel
	UserID      uint       `json:"user_id" gorm:"index:idx_tasks_user_id;not null"`
	Title       string     `json:"title" gorm:"type:varchar(200);not null"`
	Description string     `json:"description" gorm:"type:text"`
	StatusID    int64      `json:"status_id" gorm:"index:idx_tasks_status_id;not null"`
	Priority    string     `json:"priority" gorm:"not null;default:medium"`
	DueDate     *time.Time `json:"due_date" gorm:"index:idx_tasks_due_date"` // SQL側でNULL制約をしているためポインタで指定

	// リレーション
	User   *User       `gorm:"foreignKey:UserID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"user,omitempty"`
	Status *TaskStatus `gorm:"foreignKey:StatusID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"status,omitempty"`
}

File: frontend/src/types/task.ts
export interface Task {
  id: number
  user_id: number
  title: string
  description: string
  status_id: number
  priority: 'low' | 'medium' | 'high'
  due_date: string | null
  created_at: string
  updated_at: string
}

export interface CreateTaskBody {
  title: string
  description?: string
  status_id?: number
  priority?: 'low' | 'medium' | 'high'
  due_date?: string
}

export interface UpdateTaskBody {
  title: string
  description?: string
  status_id?: number
  priority?: 'low' | 'medium' | 'high'
  due_date?: string
}
